context.go

```diff
@@ -234,10 +247,12 @@ func (bwss *blipWebsocketServer) handshake(w http.ResponseWriter, r *http.Reques
 	}
 
 	bwss.blipCtx.activeSubProtocol = extractAppProtocolId(protocol)
+
+	close(bwss.wsHandshakeCompleted)
```

Comment:
Whats' the behavior if we hit an unsupported protocol or `websocket.Accept` fails?

Should we close the wsHandshakeCompleted or somehow cancel ctx for `waitUntilStarted`? The code in description will leak a go routine if this fails?

Reply:
I had thought `h.ctx()` would be cancelled when the WS upgrade fails, but I can tweak this to be a bit more robust.

Reply:
Changed to defer and value based on `err == nil` to catch early returns